### ISSUE TITLE
Top navigation hamburger menu items don't wrap text when menu item has long label. 

### ISSUE THREAD

**Issue by DBD324** [mi6/ic-design-system#701]
## Summary of the bug

Long menu item labels overflow the edge of the screen causing visual information to be lost. This constitutes an accessibility fail due to loss of content. 

## 🪜 How to reproduce

Tell us the steps to reproduce the problem:

1. Go to page: Website homepage
2. Change browser width to force small breakpoint
3. Open hamburger menu
4. Change menu item text to longer label. 
5. See overflow issue

## 📸 Screenshots or code

If you can, add screenshots to show the problem. If you need to show a code example, you can include a snippet or link to a repository with an example.

![image](https://github.com/mi6/ic-design-system/assets/116820773/34b12c69-de11-4ea3-9419-4917858ca2ee)

## 🖥 📱 Device

- Type: All
- Device: All
- OS version: All
- Browser version: All

## 🧐 Expected behaviour

When presented in the hamburger menu, menu items should wrap their text and expand in height so that the full label is visible. 

The selection indicator would also need to grow to the correct menu item height. 

Additionally, if something else cause the text size to change, such as a user-controlled setting, then the menu item should always grow to contain its content and maintain its padding. Need to test for text resizing and lineheight changes to ensure no overflow problems occur. 

On larger breakpoints when the nav item is shown in the nav bar, the nav item should grow horizontally to contain its label. Side scrolling nav bar should then occur if required. 

## 📝 Acceptance Criteria

If relevant, describe in full detail the different interactions and edge cases that the component or patterns needs to fulfil.

_Given_ that a top navigation is displayed on a xs, sm, or md breakpoint with hamburger menu
_When_  a hamburger menu item's label is longer than can be displayed in the available width of the menu
_Then_ the text label should wrap onto multiple and not overflow the page boundary
_and_ the menu item height should grow to maintain its top and bottom padding
_and_ the selection indicator should also grow to be the full-height of the menu item. 

_Give_ that t op nav is displayed on xs,sm,md breakpoint with hamburger menu, 
_When_ a user changes the text size or line-height (or something else that affects the height of the text label)
_Then_ the menu item should grow vertically to contain it and not have overflow in any direction. 

## Additional info

Tell us anything else useful to help us fix or understand the problem.
